src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -3481,6 +3481,13 @@ static QVariant fcnGetFeature( const QVariantList &values, const QgsExpressionCo
   }
 
   const QVariant &attVal = values.at( 2 );
+
+  const QString cacheValueKey = QStringLiteral( "getfeature:%1:%2:%3" ).arg( vl->id() ).arg( attributeId ).arg( attVal.toString() );
```

Comment:
Wouldn't a single multiattribute `.arg` call be faster?

Reply:
You can't when you have any non-string arguments

Reply:
`QString::number( fid )`?